packages/utils/src/common/promise-all.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Promise.allSettled with error handling, safe alternative to Promise.all
+ * @param promises
+ */
+export async function promiseAll<T = unknown>(
+  promises: Promise<T>[]
+): Promise<T[]> {
+  const states = await Promise.allSettled(promises)
+  const rejected = states.filter((state) => state.status === "rejected")
+  if (rejected.length) {
+    throw new Error((rejected[0] as PromiseRejectedResult).reason)
+  }
+
+  return states.map((state) => (state as PromiseFulfilledResult<T>).value)
+}
```

Comment:
Looks good to me I just have a small suggestion to prevent having to do all that typecasting as you have currently had to do in the order service.

```suggestion
export async function promiseAll<T = unknown>(
promises: (T | PromiseLike<T>)[]
): Promise<T[]> {
  const states = await Promise.allSettled(promises)
  const rejected = states.filter((state) => state.status === "rejected")
  if (rejected.length) {
    throw new Error((rejected[0] as PromiseRejectedResult).reason)
  }

  return states.map((state) => (state as PromiseFulfilledResult<T>).value)
}
```

Reply:
I realise that what you have changed is more correct, but this way it might be easier to swap out the use of Promise.all a cross the entire code base without having to modify anything else 😄

Reply:
ahah I ll add that in 👍